### DISCUSSION TITLE
Strip directives off DOM after initializing

### DISCUSSION THREAD

**Discussion by oliverw** [alpinejs/alpine#3892] [3. Feature ideas]
This is of purely cosmetical nature of course but some detail I like about vue and petite-vue is that directives are completely stripped off the DOM after components are initialized. I would like to see this with AlpineJS too.

**Comment by ekwoka** [comment]
It would be a pretty low priority, mainly because users won't look at it, and they can be useful for debugging, but also because right now, Alpine watches for directives to be added and removed. While likely not used a ton by people, you can actually use other things to add attribute nodes to elements, or even just write it directly in the html tree in the browser. Useful to work with other tools and to debug.

**Reply by oliverw** [reply]
Understood. Like I said. Nothing of importance.
